src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/SymbolMatcherTests.cs

```diff
@@ -1847,6 +1847,69 @@ static void Main()
             Assert.Equal("<>9__0_0", field1.Name);
         }
 
+        [Fact]
+        public void SynthesizedDelegates_03()
+        {
+            // This test checks that lambdas between source0 and source1 are not the same,
+            // as they differ in default parameter value or `params` modifier.
+            var source0 = """
+                class C
+                {
+                    unsafe static void F()
+                    {
+                        var _1 = (int a, int b = 1) => a;
+                        var _2 = (int* x, int[] c) => c;
+                    }
+                }
+                """;
+            var source1 = """
+                class C
+                {
+                    unsafe static void F()
+                    {
+                        var _1 = (int a, int b = 2) => a;
+                        var _2 = (int* x, params int[] c) => c;
+                    }
+                }
+                """;
+
+            var compilation0 = CreateCompilation(source0, options: TestOptions.UnsafeDebugDll);
+            var compilation1 = CreateCompilation(source1, options: TestOptions.UnsafeDebugDll);
+
+            var peRef0 = compilation0.EmitToImageReference();
+            var peAssemblySymbol0 = (PEAssemblySymbol)CreateCompilation("", new[] { peRef0 }).GetReferencedAssemblySymbol(peRef0);
+            var peModule0 = (PEModuleSymbol)peAssemblySymbol0.Modules[0];
+
+            var reader0 = peModule0.Module.MetadataReader;
+            var decoder0 = new MetadataDecoder(peModule0);
+
+            PEDeltaAssemblyBuilder.GetAnonymousTypeMapFromMetadata(reader0, decoder0, out _, out var anonymousDelegates0);
+            Assert.Equal(2, anonymousDelegates0.Count);
+            Assert.Equal("<>f__AnonymousDelegate0", anonymousDelegates0["<>f__AnonymousDelegate0"].Type.ToString());
+            Assert.Equal("<>f__AnonymousDelegate1", anonymousDelegates0["<>f__AnonymousDelegate1"].Type.ToString());
+
+            var testData = new CompilationTestData();
+            compilation1.EmitToArray(testData: testData);
+            var peAssemblyBuilder = (PEAssemblyBuilder)testData.Module;
+
+            var c = compilation1.GetMember<NamedTypeSymbol>("C");
+            var displayClass = peAssemblyBuilder.GetSynthesizedTypes(c).Single();
+            Assert.Equal("<>c", displayClass.Name);
+
+            var emitContext = new EmitContext(peAssemblyBuilder, null, new DiagnosticBag(), metadataOnly: false, includePrivateMembers: true);
+
+            var fields = displayClass.GetFields(emitContext).ToArray();
+            var field1 = fields[1];
+            var field2 = fields[2];
+            Assert.Equal("<>9__0_0", field1.Name);
+            Assert.Equal("<>9__0_1", field2.Name);
+
+            var matcher = new CSharpSymbolMatcher(null, null, anonymousDelegates0, compilation1.SourceAssembly, emitContext, peAssemblySymbol0);
+
+            Assert.Null(matcher.MapDefinition(field1));
```

Comment:
>Assert.Null(matcher.MapDefinition(field1)); [](http://example.com/codeflow?start=12&length=43)

Do we have a positive test? I.e. for types that should be mapped to each other even in presence of default values and `params`?

Reply:
Good catch, we didn't. Added. And it actually didn't work due to missing `ParamArrayAttribute`, but after merging #65533, it works.